static/js/chart/draw.ts

```diff
@@ -21,9 +21,47 @@ import {
   DataPoint,
 } from "./base";
 
+import {
+  randDomId,
+} from "./../util";
+
 const NUM_X_TICKS = 5;
 const NUM_Y_TICKS = 5;
 const MARGIN = { top: 20, right: 10, bottom: 30, left: 35, yAxis: 3 };
+// The middleAdjustment here is to avoid line being like underscore, and move the line into middle place.
+const LEGEND = {ratio: 0.2, minWidth: 150, height: 20, middleAdjustment: 5, leftMarginForChart: 50 };
+
+/**
+ * Return an array of dashes given a string array.
+ *
+ * @param labels
+ */
+function getDashes(labels: string[]) {
+  // Line dash styles
+  const DASHES = ['', '5, 5', '10, 5', '5, 10', '1, 5', '5, 1', '0.9', '5, 5, 1, 5'];
+  let dashes : string[];
+  dashes = [];
+  if (labels.length <= 8) {
```

Comment:
how about do not use the special case but only using the generic way, so move the getDashesHelper to getDashes function. Also in dev.js can you make a chart that draw 20 lines using this so we can see how the style looks like.

Sorry for adding this requirements. Want to make sure it would look right.

Reply:
The reason i am using special case is I am wondering to follow the first line to be solid, then 5, 5. Definitely can test this function in dev.js. But do I need to remove the special case as mostly we are using special case?

Reply:
We can special case the single solid line perhaps, and use the helper for the rest?

getDashesHelper is a good case for a unit test, so that we can test the values for the first n or so.

Also, instead of 8, use DASHES.length which will continue working as you change the array.